pkg/parser/additional/certificate_utils.go

```diff
@@ -0,0 +1,79 @@
+package additional
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+// CheckCertificateBody verifies if the attribute 'certificate_body' refers a file
+func CheckCertificateBody(content string) (b bool, s string) {
+	var re = regexp.MustCompile(`[0-9a-zA-Z-/\\_]+\.pem`)
+	match := re.FindAllString(content, -1)
+
+	if match == nil {
+		return false, ""
+	}
+
+	return true, match[0]
+}
+
+func getCertificateInfo(filePath string) (a [3]int, i int) {
+	certPEM, err := os.ReadFile(filePath)
+
+	if err != nil {
+		panic("Failed to read the file " + filePath)
+	}
+
+	block, _ := pem.Decode(certPEM)
+	if block == nil {
+		panic("Failed to parse the certificate PEM " + filePath)
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		panic("Failed to parse the certificate" + filePath)
+	}
+
+	var date [3]int
+	date[0] = cert.NotAfter.Year()
+	date[1] = int(cert.NotAfter.Month())
+	date[2] = cert.NotAfter.Day()
+
+	var rsaBytes int
+
+	switch t := cert.PublicKey.(type) {
+	case *rsa.PublicKey:
+		_ = t
```

Comment:
If you are not using t for anything, I think you can change this to `switch cert.PublicKey.(type) {` and remove `_ = t`

Reply:
It was my first approach, but 'golangci-lint' seems not to like it 😆 If I remove 't', 'golangci-lint' returns " typeSwitchVar: 1 case can benefit from type switch with assignment (gocritic)"